What is the total loss from the normal (non-mass) examples in this example dataset?

Please use the natural logarithm in your calculation. When you use numpy.log, this is using the natural logarithm. Also, to get the total loss, please add up the losses from each ‘normal’ example.

Example	P(positive)
P1 Normal	0.6
P3 Normal	0.3
P5 Mass	0.4

In [18]:
import numpy as np

y_true = np.array(
        [[0],
         [0]])
print(f"y_true: \n{y_true}")

y_true: 
[[0]
 [0]]


In [19]:
y_pred = np.array(
        [[0.6],
         [0.3]])
print(f"y_pred: \n{y_pred}")

y_pred: 
[[0.6]
 [0.3]]


In [20]:
loss_reg_1 = -1 * np.sum(y_true * np.log(y_pred)) + \
                -1 * np.sum((1 - y_true) * np.log(1 - y_pred))
print(f"loss_reg_1: {loss_reg_1:.4f}")

loss_reg_1: 1.2730


# Answer

Since these are negative examples, the losses will be  -log(1-P(positive))−log(1−P(positive)).

For P1, -log(1-0.6) = 0.91−log(1−0.6)=0.91.

For P3 -log(1-0.3) = 0.36−log(1−0.3)=0.36.

The sum is 0.91 + 0.36 = 1.270.91+0.36=1.27.